build.nix

```diff
@@ -0,0 +1,130 @@
+let pkgs = import ./nix;
+    hlib = pkgs.haskell.lib;
+    withCleanedPath = drv:
+      hlib.overrideCabal drv (old: {
+        src = pkgs.lib.cleanSourceWith {
+          src = old.src;
+          filter = path: type:
+            let baseName = baseNameOf (toString path);
+            in baseName != "dist";
+        };
+      });
+
+    # Mapping from package -> executbale
+    executablesMap = {
+      brig = ["brig" "brig-index" "brig-integration" "brig-schema"];
+      cannon = ["cannon"];
+      cargohold = ["cargohold" "cargohold-integration"];
+      federator = ["federator" "federator-integration"];
+      galley = ["galley" "galley-integration" "galley-schema" "galley-migrate-data"];
+      gundeck = ["gundeck" "gundeck-integration" "gundeck-schema"];
+      proxy = ["proxy"];
+      spar = ["spar" "spar-integration" "spar-schema" "spar-migrate-data"];
+      stern = ["stern"];
+
+      bonanza = ["bonanza"];
+      billing-team-member-backfill = ["billing-team-member-backfill"];
+      api-smoketest = ["api-smoketest"];
+      zauth = ["zauth"];
+    };
+
+    attrsets = pkgs.lib.attrsets;
+    hPkgs = pkgs.haskell.packages.ghc8107.override {
+      overrides = hself: hsuper:
+        let externalOverrides = import ./nix/haskell-overrides.nix hsuper hself;
+            localOverrides = import ./nix/local-overrides.nix hself hsuper;
+            manualOverrides = {
+              network-arbitrary = hlib.markUnbroken (hlib.doJailbreak hsuper.network-arbitrary);
+              cql = hlib.markUnbroken hsuper.cql;
+              cql-io =  hlib.markUnbroken (hlib.dontCheck hsuper.cql-io);
+              lens-datetime = hlib.markUnbroken (hlib.doJailbreak hsuper.lens-datetime);
+              wai-predicates = hlib.markUnbroken hsuper.wai-predicates;
+              bytestring-arbitrary = hlib.markUnbroken (hlib.doJailbreak hsuper.bytestring-arbitrary);
+              invertible = hlib.markUnbroken hsuper.invertible;
+              polysemy-check = hlib.markUnbroken (hlib.doJailbreak hsuper.polysemy-check);
+              swagger = hlib.doJailbreak externalOverrides.swagger;
+              multihash = hlib.doJailbreak externalOverrides.multihash;
+              wire-message-proto-lens = hlib.addBuildTool localOverrides.wire-message-proto-lens pkgs.protobuf;
+              types-common-journal = hlib.addBuildTool localOverrides.types-common-journal pkgs.protobuf;
+              hashable = hsuper.hashable_1_4_0_2;
```

Comment:
where do the versions overriden here come from? did you arrive them by trail-and-error?

Reply:
Yeah, whenever I saw an error saying, hey I want a newer `<package>`, I checked in  `haskell.packages.ghc8107` whether there was some `<package>_<version>`, if I found it, I just used it.